src/predictor/predictor.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 CodeMaker AI Inc. All rights reserved.
+
+import * as vscode from 'vscode';
+import { Configuration } from '..//configuration/configuration';
+import { isFileSupported } from '../utils/languageUtils';
+import CodeMakerService from '../service/codemakerService';
+
+const PredictiveGenerationCoolDown = 1000 * 30;
+
+export class Predictor {
+
+    private readonly codeMakerService: CodeMakerService;
+
+    private readonly lastPredictiveGeneration: Map<string, Date>;
+
+    constructor(codeMakerService: CodeMakerService) {
```

Comment:
I didn't find where this constructor is called, will there be another commit?

Reply:
I updated the code to glue everything together.